lms/djangoapps/edraak_university/tests/test_csv_export.py

```diff
@@ -0,0 +1,147 @@
+                course_key=unicode(self.course.id),
+                university_id='2011A-500',
+                section_number='K',
+            )
+
+    def create_rows(self):
+        """
+        Creates CSV rows for the provided emails.
+        """
+        result_store = Mock(rows=[])
+        current_task = Mock(update_state=Mock())
+
+        def csv_rows_interceptor(rows, *_args, **_kwargs):
+            """
+            Mock the upload_csv_to_report_store to return the csv rows.
+            """
+            result_store.rows = rows
+
+        for i, email in enumerate(self.EMAILS):
+            self.create_student('student{0}'.format(i), email)
+
+        with patch('instructor_task.tasks_helper.upload_csv_to_report_store', side_effect=csv_rows_interceptor):
+            with patch('instructor_task.tasks_helper._get_current_task') as mock_current_task:
+                mock_current_task.return_value = current_task
+                upload_grades_csv(None, None, self.course.id, None, 'graded')
+
+        return result_store.rows
+
+    @patch.dict(settings.FEATURES, {'EDRAAK_UNIVERSITY_CSV_EXPORT': False})
+    @ddt.data(*UNIVERSITY_ID_COLUMNS)
+    def test_headers_disabled(self, header):
+        rows = self.create_rows()
+
+        self.assertIn('email', rows[0])  # Sanity check for the header row
+        self.assertIn('username', rows[0])  # Sanity check
+        self.assertFalse(is_csv_export_enabled_on_course(self.course))  # Sanity check
+
+        self.assertEquals(len(rows[0]), len(rows[1]),
+                          'The headers and data should have the same number of columns')
+
+        self.assertNotIn(header, rows[0])  # Should not have the University ID headers if the feature is disabled
+
+    @patch.dict(settings.FEATURES, {'EDRAAK_UNIVERSITY_CSV_EXPORT': True})
+    @ddt.data(*UNIVERSITY_ID_COLUMNS)
+    def test_headers_enabled(self, header):
+        rows = self.create_rows()
+
+        self.assertIn('email', rows[0])  # Sanity check
+        self.assertIn('username', rows[0])  # Sanity check
+        self.assertTrue(is_csv_export_enabled_on_course(self.course))  # Sanity check
+
+        self.assertEquals(len(rows[0]), len(rows[1]),
+                          'The headers and data should have the same number of columns')
+
+        self.assertIn(header, rows[0])  # Should have the University ID headers when the feature is enabled
+
+    def get_row_as_dict(self, rows, index):
+        """
+        Allow accessing the row using the header name instead of integer index.
+
+         e.g. `row['Full Name']` instead of `row[10]`
+        """
+        header_row = rows[0]
+        data_row = rows[index]
+
+        return dict(zip(header_row, data_row))
+
+    @patch.dict(settings.FEATURES, {'EDRAAK_UNIVERSITY_CSV_EXPORT': True})
+    def test_with_university_id(self):
+        rows = self.create_rows()
+        row = self.get_row_as_dict(rows, index=1)
+
+        user = User.objects.get(email=self.EMAILS[0])
```

Comment:
this could create problems in the future, if the EMAILS[0] was not as well an email with id, so I think using dictionary for emails would be more expressive.